src/lib/rc/dsm.cpp

```diff
@@ -152,14 +188,15 @@ dsm_guess_format(bool reset)
 
 		uint8_t *dp = &dsm_frame[2 + (2 * i)];
 		uint16_t raw = (dp[0] << 8) | dp[1];
-		unsigned channel, value;
+		uint16_t channel = 0;
+		uint16_t value = 0;
 
 		/* if the channel decodes, remember the assigned number */
-		if (dsm_decode_channel(raw, 10, &channel, &value) && (channel < 31)) {
+		if (dsm_decode_channel(raw, 10, channel, value) && (channel < 31)) {
```

Comment:
@dagar , you didn't create the magic numbers 10 and 31, but I had to stop to figure out their legitimacy.

Reply:
It's 10 bit (1024) or 11 bit (2048) mode for DSM. Nearly everything we actually use these days is 11 bit. 

We should do a followup pass and make the modes a class enum.